File: frameworks/Java/kora/kora-jdbc/src/main/java/io/koraframework/benchmark/util/QueryUtils.java
package io.koraframework.benchmark.util;

import java.util.concurrent.ThreadLocalRandom;

public final class QueryUtils {

    private QueryUtils() { }

    private static final int MIN_QUERIES = 1;
    private static final int MAX_QUERIES = 500;
    private static final int WORLD_ROWS = 10000;

    public static int parseCount(Integer value) {
        if (value == null) {
            return MIN_QUERIES;
        }
        if (value < MIN_QUERIES) {
            return MIN_QUERIES;
        }
        return Math.min(value, MAX_QUERIES);
    }

    public static int randomWorld() {
        return ThreadLocalRandom.current().nextInt(WORLD_ROWS) + 1;
    }
}

File: frameworks/Kotlin/kora/kora-jdbc-coroutines/src/main/kotlin/io/koraframework/benchmark/util/QueryUtils.kt
package io.koraframework.benchmark.util

import java.util.concurrent.ThreadLocalRandom

object QueryUtils {
    private const val MIN_QUERIES = 1
    private const val MAX_QUERIES = 500
    private const val WORLD_ROWS = 10_000

    fun parseCount(value: Int?): Int {
        if (value == null || value < MIN_QUERIES) {
            return MIN_QUERIES
        }
        return minOf(value, MAX_QUERIES)
    }

    fun randomWorld(): Int = ThreadLocalRandom.current().nextInt(WORLD_ROWS) + 1
}
